src/Utils/RMC/RMC.as

```diff
@@ -32,14 +32,26 @@ class RMC
         if (RMC::IsRunning && (UI::IsOverlayShown() || (!UI::IsOverlayShown() && PluginSettings::RMC_AlwaysShowBtns))) {
             if (UI::RedButton(Icons::Times + " Stop RM"+lastLetter))
             {
-#if DEPENDENCY_CHAOSMODE
-                ChaosMode::SetRMCMode(false);
-#endif
+                RMC::EndTimeCopyForSaveData = RMC::EndTime;
+                RMC::StartTimeCopyForSaveData = RMC::StartTime;
                 RMC::IsRunning = false;
                 RMC::ShowTimer = false;
                 RMC::StartTime = -1;
                 RMC::EndTime = -1;
                 @MX::preloadedMap = null;
+
+#if DEPENDENCY_CHAOSMODE
+                ChaosMode::SetRMCMode(false);
+#endif
+                int secondaryCount = RMC::selectedGameMode == RMC::GameMode::Challenge ? BelowMedalCount : RMC::Survival.Skips;
+                if (RMC::GoalMedalCount != 0 || secondaryCount != 0 || RMC::GotBelowMedalOnCurrentMap || RMC::GotGoalMedalOnCurrentMap) {
+                    Renderables::Add(SaveRunQuestionModalDialog());
```

Comment:
Can you add a Plugin Setting (in `src/Settings/RMCParameters.as`) to auto save run on exit and set it to true by default ? so we don't have to open this dialog window everytime we click on stop button